src/utils.js

```diff
@@ -79,6 +83,16 @@ export async function blobToDataUrl(blob) {
   });
 }
 
+export function getFileResourcePath(href, recode = false) {
+  const path = recode ? decodeURIComponent(encodeURIComponent(href)) : href;
```

Comment:
I feel that we need to reconsider this `decodeURIComponent(encodeURIComponent(href))` logic at some point.